designer/client/src/components/graph/GraphStyled.ts

```diff
@@ -1,100 +1,92 @@
-            }
-            &:hover .connection-wrap {
-                opacity: 0.5;
-                stroke-opacity: 0.5;
-            }
-
-            &.${dragHovered} {
-                .connection {
-                    opacity: 0.5;
-                }
-                .connection-wrap {
-                    opacity: 0.5;
-                    stroke-opacity: 0.5;
-                    stroke-width: 24;
-                    stroke-linecap: square;
-                    stroke: ${theme.custom.colors.red};
-                }
-            }
-
-            .connection {
-                stroke: ${theme.custom.colors.primaryColor};
-                stroke-width: 2;
-            }
-
-            /*simple method to get dragging state*/
-            &[style*="pointer"] {
-                .connection {
-                    stroke-width: 3;
-                    stroke-dasharray: 3 0 3;
-                }
-            }
-
-            .link-tool {
-                /*TODO: fix this without css*/
-                &[visibility="hidden"] {
-                    /*remove hidden tools from size calculations*/
-                    display: none;
-                }
-                .tool-remove {
-                    cursor: default;
-                    filter: saturate(0.75);
-                    transition: all 0.25s ease-in-out;
-                    &:hover {
-                        filter: saturate(2);
-                    }
-                }
-            }
-        }
-
-        .joint-type-esp-model {
-            .body .joint-port-body .background {
-                transition: all 0.25s ease-in-out;
-            }
-        }
-    `,
-);
+export const GraphStyled = styled(FocusableStyled)(({ theme }) => ({
+    minHeight: "300px",
+    minWidth: "300px",
+    color: theme.palette.common.white,
+    ".arrowMarker path": {
+        fill: theme.palette.text.secondary,
+    },
+    " @media (hover: none)": {
+        ".joint-theme-default .joint-link .link-tool &:hover .connection-wrap": {
+            display: "none",
+            ".marker-arrowhead": {
+                opacity: 0,
+                transform: "scale(0.3)",
+            },
+        },
+    },
+    ".joint-theme-default.joint-link": {
+        ":hover": {
+            ".connection": {
+                stroke: theme.palette.primary.main,
+            },
+        },
+        ".marker-arrowhead": {
+            // small element with big stroke allows arrow to overflow link path
+            transform: "scale(0.1)",
+            strokeWidth: 45,
+            fill: theme.palette.primary.main,
+            stroke: theme.palette.primary.main,
+            transition: "all 0.25s ease-in-out",
+            "&:hover": {
+                fill: theme.palette.primary.main,
+                stroke: theme.palette.primary.main,
+            },
+        },
+        ".marker-arrowhead-group-source": {
+            display: "none",
+        },
+        ".connection-wrap": {
+            "&:hover": {
```

Comment:
The cursor should be "default" outside of `&hover` because when the cursor is at the edge (very small space) around the connection it gets defaultly set to "move"
![image](https://github.com/TouK/nussknacker/assets/92804917/5a5c8768-81fe-484f-8194-18b1ce74a27b)